etl/kpinetwork_analizer.py

```diff
@@ -82,6 +82,147 @@
+
+    dynamic_frame = glueContext.create_dynamic_frame.from_options(
+        connection_type="postgresql",
+        connection_options={
+            "url": "{}/{}".format(catalog_connection["url"], database),
+            "user": catalog_connection["user"],
+            "password": catalog_connection["password"],
+            "dbtable": db_table,
+        },
+    )
+    dynamic_frame = dynamic_frame.toDF().createOrReplaceTempView(
+        "{}_table".format(db_table)
+    )
+    sqldf = spark.sql("SELECT id FROM {}_table".format(db_table))
+    result_from_sqldf = sqldf.collect()
+    ids_from_db = [row.id for row in result_from_sqldf]
+    return ids_from_db
+
+
+def exists_metric_from_file(env, company_id, scenario_name, metric_name):
+    resultant_metric_id = None
+    database = "kpinetworkdb"
+    if env == "demo":
+        database = "demokpinetworkdb"
+
+    ctg_connection = "{}_connection".format(env)
+    catalog_connection = glueContext.extract_jdbc_conf(ctg_connection)
+
+    dynamic_fs_frame = glueContext.create_dynamic_frame.from_options(
+        connection_type="postgresql",
+        connection_options={
+            "url": "{}/{}".format(catalog_connection["url"], database),
+            "user": catalog_connection["user"],
+            "password": catalog_connection["password"],
+            "dbtable": "financial_scenario",
+        },
+    )
+
+    dynamic_fs_frame = dynamic_fs_frame.toDF().createOrReplaceTempView(
+        "financial_scenario_table"
+    )
+    fs_dataframe = spark.sql(
+        "SELECT id FROM financial_scenario_table WHERE company_id = '{}' AND name = '{}'".format(
+            company_id, scenario_name
+        )
+    )
+    result_fs_dataframe = fs_dataframe.collect()
+    fs_ids = [row.id for row in result_fs_dataframe]
+
+    if len(fs_ids) > 0:
+        financial_scenario_id = fs_ids[0]
+        dynamic_sm_frame = glueContext.create_dynamic_frame.from_options(
+            connection_type="postgresql",
+            connection_options={
+                "url": "{}/{}".format(catalog_connection["url"], database),
+                "user": catalog_connection["user"],
+                "password": catalog_connection["password"],
+                "dbtable": "scenario_metric",
+            },
+        )
+
+        dynamic_sm_frame = dynamic_sm_frame.toDF().createOrReplaceTempView(
+            "scenario_metric_table"
+        )
+        sm_dataframe = spark.sql(
+            "SELECT metric_id FROM scenario_metric_table WHERE scenario_id = '{}'".format(
+                financial_scenario_id
+            )
+        )
+        result_sm_dataframe = sm_dataframe.collect()
+        sm_ids = [row.metric_id for row in result_sm_dataframe]
+
+        if len(sm_ids) > 0:
+            sc_metric_ids = sm_ids
+            for sc_metric_id in sc_metric_ids:
+                metric_id = sc_metric_id
+                dynamic_m_frame = glueContext.create_dynamic_frame.from_options(
```

Comment:
You have three read table functionality, and the only argument different for each one is the "dbtable", so you should create a function to read values from the database if you don't have it.

Reply:
I have created the function **read_table_from_database**.